src/jquery.uls.core.js

```diff
@@ -225,30 +264,26 @@
 				this.$menu.on( 'keydown', $.proxy( this.keypress, this ) );
 			}
 
-			languagesCount = Object.keys( this.options.languages ).length;
-			lcd = this.$resultsView.lcd( {
-				languages: this.languages,
-				columns: columnsOptions[ this.getMenuWidth() ],
-
-				quickList: languagesCount > 12 ? this.options.quickList : [],
-				clickhandler: $.proxy( this.select, this ),
-				source: this.$languageFilter,
-				showRegions: this.options.showRegions,
-				languageDecorator: this.options.languageDecorator,
-				noResultsTemplate: this.options.noResultsTemplate
-			} ).data( 'lcd' );
+			this.createLanguageFilter();
 
-			this.$languageFilter.languagefilter( {
-				$target: lcd,
-				languages: this.languages,
-				searchAPI: this.options.searchAPI,
-				onSelect: $.proxy( this.select, this )
-			} );
-
-			this.$languageFilter.on( 'noresults.uls', $.proxy( lcd.noResults, lcd ) );
 			this.$languageFilter.on( 'resultsfound.uls', $.proxy( this.success, this ) );
 
 			$( 'html' ).click( $.proxy( this.cancel, this ) );
+			$( window ).resize( $.debounce( 250, this.resize.bind( this ) ) );
+		},
+
+		resize: function () {
+			var menuWidth = this.getMenuWidth();
+
+			if ( this.menuWidth === menuWidth ) {
+				return;
+			}
+
+			this.menuWidth = menuWidth;
+			this.shouldRecreate = true;
+			if ( !this.shown ) {
```

Comment:
Shouldn't the ULS menu resize when shown?

Reply:
As @Nikerabbit's sharp eye noticed, we would have problems with recreating LCD while user has typed search query.

I am considering adding hot swapping of LCD options in the future, without the need to recreate the whole thing. Haven't done assessment what will need to be changed, but I think we'll need to change the core logic.